### PR TITLE
Adding shortcuts to remarks glossary in create agenda items page (TM-2717)

### PR DESCRIPTION
**Pull request by pveland621**

### CODE REVIEW

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx [hunk 1]
```diff
@@ -80,11 +85,15 @@ const RemarksGlossary = ({ onRemarkClick, remarks, remarkCategories }) => {
         }}
       />
       <div className="remarks-glossary-container">
+        <div className="usa-grid-full remarks-categories-container">
+          {remarkCategories$.map(a => (
+            <a className="remarks-category-container a" tabIndex={0} role="button" onClick={() => processClick(a)}>{a.desc_text}</a>))}
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:90]: nice!

**Comment by SidelineCory24** [comment]
I would update the styling so that the rows format evenly on different screen sizes/edge cases. Checkout how we use `@media screen and min-width/max-width` in the code.

image

**Comment by elizabeth-jimenez** [comment]
> I would update the styling so that the rows format evenly on different screen sizes/edge cases. Checkout how we use `@media screen and min-width/max-width` in the code.
> 
> image
> 
> image
> 
> something like this for the screen size edge cases and update the styling accordingly `@media screen and (min-width: 2070px) and (max-width: 2445px)` `@media screen and (max-width: 2860px)`

Nice catch, Cory!
Another potential solution, is adding the following to `remarks-categories-container`
```
display: grid;
grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
```

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx (hunk 1 above)
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:90]: you don't seem to ever use these classes: `remarks-category-container a` 🧼

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx [hunk 2]
```diff
@@ -80,11 +85,15 @@ const RemarksGlossary = ({ onRemarkClick, remarks, remarkCategories }) => {
...
+        </div>
         {remarkCategories$.map(category => {
           const remarksInCategory = orderBy(remarks$$.filter(f => f.rc_code === category.code), 'order_num');
           return (
             <div key={category.code}>
-              <div className={`remark-category remark-category--${category.code}`}>{category.desc_text}</div>
+              <div id={`remark-category-${category.code}`} className={`remark-category remark-category--${category.code}`}>{category.desc_text}</div>
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:96]: 🙌

File: src/sass/_agendaItemMaintenance.scss [hunk 3]
```diff
@@ -324,12 +324,12 @@
   }
 
   a {
-    padding-left: 5px
+    padding: 10px
   }
```
Comment (elizabeth-jimenez) [inline src/sass/_agendaItemMaintenance.scss:328]: did you make sure this did not affect other `a` elements? could be safer to add this around line 420, where you are already targeting your specific `a`

Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:328]: you can do the `background-color: red` trick to quickly see the effect.

File: src/sass/_agendaItemMaintenance.scss [hunk 4]
```diff
@@ -409,6 +409,23 @@
...
+  flex-direction: row;
+  flex-wrap: wrap;
+  margin: 10px 10px 30px 10px;
+  justify-content: space-between;
+
+  a {
+    min-width: 200px;
+    color: black;
+    font-weight: bold;
+    cursor: pointer;
```
Comment (SidelineCory24) [inline src/sass/_agendaItemMaintenance.scss:424]: nice! 🙌

**Comment by elizabeth-jimenez** [comment]
https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2095#discussion_r858794021 still need to address?

**Comment by elizabeth-jimenez** [comment]
image

Clicking on a section titles scrolls it into view, but does not bring the title into view.

File: src/sass/_agendaItemMaintenance.scss [hunk 5]
```diff
@@ -500,6 +518,7 @@
     .remark-category--#{$remark-id} {
       background-color: $color;
       color: $color-white;
+      scroll-margin-top: 55px;
```
Comment (SidelineCory24) [inline src/sass/_agendaItemMaintenance.scss:521]: 🐐

Comment (SidelineCory24) [reply src/sass/_agendaItemMaintenance.scss:521]: @elizabeth-jimenez I know you did most of the review lift on this, but the scroll/snap looks good to me.

https://user-images.githubusercontent.com/55964163/166560759-5467c06c-e1d6-454c-a8d3-e5e6c01d7015.mp4

Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:521]: It looks great at this view 😁 but checkout the stacked view - we are gonna need a media query for that one

Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:521]: rip - i thought i pulled the latest and then checked the stacked view. It looks alright on stacked!

**Comment by elizabeth-jimenez** [comment]
I agree with Cory that the Scroll looks good - Let's just adjust it by a few pixels and add a media query:

`  @each $remark-id, $color in $remarks {
    .remark-category--#{$remark-id} {
      background-color: $color;
      color: $color-white;
      scroll-margin-top: 60px;
    }
    @media screen and (max-width: 1600px) {
      .remark-category--#{$remark-id} {
        scroll-margin-top: 32px;
      }
    }
  }`

**Review by elizabeth-jimenez — APPROVED** [review]

**Pull request merged**